sphinx_needs/api/need.py

```diff
@@ -408,8 +409,9 @@ def del_need(app: Sphinx, id) -> None:
     :param app: Sphinx application object.
     :param id: Sphinx need id.
     """
-    if id in app.env.needs_all_needs:
-        del app.env.needs_all_needs[id]
+    env = unwrap(app.env)
```

Comment:
sort of. changing the pre-commit hook to a local hook means it has access to the `sphinx` types, can can see that `needs_all_things` is not an attribute on the `sphinx.environment.BuildEnvironment` class, this leads to many additional type errors which causes the CI to fail. The changes in this PR address those additional changes. Perhaps we should discuss #586 first?

unwrap is needed because the objects can be `None`, and we need to convince the type checker that we know they are not None. `unwrap` is not the only way to do this, you can also use `typing.cast`, but it's more verbose (and won't offer any protection at runtime)

Reply:
Ok got it and I'm convinced.
For me we can go with this solution,